RadixWallet/Features/AccountRecoveryScan/Children/AccountRecoveryScanInProgress/AccountRecoveryScanInProgress.swift

```diff
@@ -210,13 +210,13 @@ public struct AccountRecoveryScanInProgress: Sendable, FeatureReducer {
 				assert(networkID == state.networkID)
 				loggerGlobal.debug("Creating accounts with networkID: \(networkID)")
 				return .run { send in
-					let accounts = try await publicHDKeys.enumerated().asyncMap { localOffset, publicHDKey in
+					let accounts = await publicHDKeys.enumerated().asyncMap { localOffset, publicHDKey in
 						let offset = localOffset + globalOffset
 						let appearanceID = await accountsClient.nextAppearanceID(networkID, offset)
-						return try Account(
+						return Account(
 							networkID: networkID,
 							factorInstance: .init(factorSourceId: id, publicKey: publicHDKey),
-							displayName: .init(rawValue: L10n.AccountRecoveryScan.InProgress.nameOfRecoveredAccount) ?? "Unnamed",
+							displayName: (try? .init(validating: L10n.AccountRecoveryScan.InProgress.nameOfRecoveredAccount)) ?? DisplayName(value: "Unnamed"),
```

Comment:
Help me understand this. We are using `DisplayName`, which basically is a struct that validates the associated String isn't empty nor longer than 30 characters. And in this case we are trying to init it with the localised version of `nameOfRecoveredAccount`, and if it fails we default to one that we know meets that criteria.

If that's the case, and the only thing we do on the `init(validating:)` is check those two conditions:
1) what would you think about making the `init?` an optional one instead of a throwing method? Maybe I am missing something, but it seems to me that every time the `try?` throws, we are gonna want to deal with a null value (as here) rather than catch an error.

2) when calling `init(nonEmpty:)`, as in the next changed file, how do we know that it meets the length criteria? In other words, why is such initializer not throwing as this one?

Reply:
`nonEmpty` takes a `NonEmptyString` which is already guaranteed to be non-empty.

There are many improvements to be made for sure... I'm trying to get my monster PR merged ASAP and we will all have to help out improving code base :)

Reply:
> nonEmpty takes a NonEmptyString which is already guaranteed to be non-empty.

Right, but if I understood correctly the `DisplayName` also validates that it isn't larger than 30 chars. Are we performing such validation?

Reply:
yes correct

Reply:
Then I don't understand why the `init(nonEmpty:)` (performs 1 validation) doesn't throw while the `init(validating:)` (performs 2 validations) does throw.

Reply:
I will force unwrap. the L10n should not be empty.

Reply:
Or rather use a non throwing (should have been hidden, but here convenient it is not) init `init:value` (memberwise init exported by UniFFI.